--- origsrc/xsane-0.995/configure.in	2007-11-21 12:57:18.000000000 -0600
+++ src/xsane-0.995/configure.in	2007-12-24 21:00:33.921875000 -0600
@@ -103,7 +103,13 @@
 
 dnl Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS(windows.h os2.h fcntl.h unistd.h libc.h sys/time.h sys/types.h zlib.h)
+AC_CHECK_HEADERS(os2.h fcntl.h unistd.h libc.h sys/time.h sys/types.h zlib.h)
+dnl Do not test for windows.h on Cygwin, only MinGW
+case "$host_os" in
+  mingw*)
+    AC_CHECK_HEADERS(windows.h)
+    ;;
+esac
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
@@ -312,4 +318,4 @@
 echo "* ... PLEASE READ SANE DOCUMENTATION BEFORE STARTING XSANE ... *"
 echo "* ------------------------------------------------------------ *"
 echo "****************************************************************"
-cat xsane.NEWS
+cat ${srcdir}/xsane.NEWS
--- origsrc/xsane-0.995/doc/Makefile.in	2006-01-01 07:08:11.000000000 -0600
+++ src/xsane-0.995/doc/Makefile.in	2007-12-24 21:00:33.921875000 -0600
@@ -57,14 +57,14 @@
 	$(MKINSTALLDIRS) $(DESTDIR)$(datadir)
 
 	$(MKINSTALLDIRS) $(DESTDIR)$(xsanedocdir)
-	@for page in *.html; do\
-	  echo installing $${page} in $(DESTDIR)$(xsanedocdir)/$${page}...; \
-	  $(INSTALL_DATA) $${page} $(DESTDIR)$(xsanedocdir)/$${page} || exit 1; \
+	@for page in $(srcdir)/*.html; do\
+	  echo installing $${page} in $(DESTDIR)$(xsanedocdir)...; \
+	  $(INSTALL_DATA) $${page} $(DESTDIR)$(xsanedocdir)/ || exit 1; \
 	done
 
-	@for image in *.jpg; do\
-	  echo installing $${image} in $(DESTDIR)$(xsanedocdir)/$${image}...; \
-	  $(INSTALL_DATA) $${image} $(DESTDIR)$(xsanedocdir)/$${image} || exit 1; \
+	@for image in $(srcdir)/*.jpg; do\
+	  echo installing $${image} in $(DESTDIR)$(xsanedocdir)...; \
+	  $(INSTALL_DATA) $${image} $(DESTDIR)$(xsanedocdir)/ || exit 1; \
 	done
 
 uninstall:
--- origsrc/xsane-0.995/intl/Makefile.in	2006-01-01 07:49:23.000000000 -0600
+++ src/xsane-0.995/intl/Makefile.in	2007-12-24 21:00:33.937500000 -0600
@@ -73,7 +73,7 @@
 .c.lo:
 	$(LIBTOOL) --mode=compile $(COMPILE) $<
 
-INCLUDES = -I.. -I. -I$(top_srcdir)/intl -I$(top_srcdir)/lib -I$(top_srcdir)/include
+INCLUDES = -I.. -I. -I$(top_srcdir)/intl -I$(top_srcdir)/lib -I$(top_builddir)/include -I$(top_srcdir)/include
 
 all: all-@USE_INCLUDED_LIBINTL@
 
--- origsrc/xsane-0.995/lib/Makefile.in	2003-11-22 03:51:42.000000000 -0600
+++ src/xsane-0.995/lib/Makefile.in	2007-12-24 21:00:33.937500000 -0600
@@ -30,7 +30,7 @@
 
 CC = @CC@
 INCLUDES = -I. -I$(srcdir) \
-	-I$(top_builddir)/include/sane -I$(top_srcdir)/include
+	-I$(top_builddir)/include/sane -I$(top_builddir)/include -I$(top_srcdir)/include
 CPPFLAGS = @CPPFLAGS@
 CFLAGS = @CFLAGS@
 LDFLAGS = @LDFLAGS@
--- origsrc/xsane-0.995/m4/msnative-struct.m4	2004-05-13 17:13:01.000000000 -0500
+++ src/xsane-0.995/m4/msnative-struct.m4	2007-12-24 21:00:33.953125000 -0600
@@ -12,8 +12,8 @@
 dnl Add ms native struct gcc option to CFLAGS if available
 dnl
 dnl  if test x"$glib_native_win32" = xyes; then
- AC_CHECK_HEADERS(windows.h,
- [
+case "$host_os" in
+  mingw*)
     if test x"$GCC" = xyes; then
       msnative_struct=''
       AC_MSG_CHECKING([how to get MSVC-compatible struct packing])
@@ -42,7 +42,6 @@
         AC_MSG_RESULT([${msnative_struct}])
       fi
     fi
-dnl  fi
- ])
+  esac
 ])
 
--- origsrc/xsane-0.995/src/Makefile.in	2007-02-24 11:55:46.000000000 -0600
+++ src/xsane-0.995/src/Makefile.in	2007-12-24 21:00:33.968750000 -0600
@@ -70,12 +70,11 @@
 	$(INSTALL_DATA) $(srcdir)/xsane-style.rc $(DESTDIR)$(sanedatadir)/xsane/xsane-style.rc
 	$(INSTALL_DATA) $(srcdir)/xsane-startimage.pnm $(DESTDIR)$(sanedatadir)/xsane/xsane-startimage.pnm
 	$(INSTALL_DATA) $(srcdir)/xsane-calibration.pnm $(DESTDIR)$(sanedatadir)/xsane/xsane-calibration.pnm
-	$(INSTALL_DATA) $(srcdir)/xsane-logo.xpm $(DESTDIR)$(sanedatadir)/xsane/xsane-logo.xpm
 	$(INSTALL_DATA) $(srcdir)/xsane-gpl.txt $(DESTDIR)$(sanedatadir)/xsane/xsane-gpl.txt
 	$(INSTALL_DATA) $(srcdir)/xsane-eula.txt $(DESTDIR)$(sanedatadir)/xsane/xsane-eula.txt
-	@for logo in *-logo.xpm; do \
-	  echo installing $(DESTDIR)$(sanedatadir)/xsane/$${logo}; \
-	  $(INSTALL_DATA) $(srcdir)/$${logo} $(DESTDIR)$(sanedatadir)/xsane/$${logo}; \
+	@for logo in $(srcdir)/*-logo.xpm; do \
+	  echo installing $(DESTDIR)$(sanedatadir)/xsane/$${logo} into $(DESTDIR)$(sanedatadir)/xsane/; \
+	  $(INSTALL_DATA) $${logo} $(DESTDIR)$(sanedatadir)/xsane/; \
 	done 
 
 uninstall:
